### DISCUSSION TITLE
I had to reinstall Docker after a problem. now my library is no longer visible in immich

### DISCUSSION THREAD

**Discussion by Pouletgrill** [immich-app/immich#8040] [Q&A] [answered]
Hello, I had to reinstall Docker after a problem whose only solution was apparently reinstallation. 
Subsequently I redid the "docker compose up -d" in my yml directory. now my library is no longer visible in immich and my images all seem to be in "untracks files". Can you help me restore everything? at least I know that my library is not really empty and that I have a backup. thank you.

**Comment by JW-CH** [comment]
Show us your docker compose or logs. Otherwise helping you is nearly impossible.

**Reply by Pouletgrill** [reply]
yes of course, i was not sure what info was needed to investigate the problem. My guess is that by reinstalling Docker, i might have lost my database and now imich is not aware of what to do with my librairy full of unlinked data. 
[docker-compose.yml.txt](https://github.com/immich-app/immich/files/14636877/docker-compose.yml.txt)

i joint my docker compose file here (i had to made it a ".txt" for it to pass on github)
for the log i dont know where to find it

**Reply by bo0tzz** [reply]
Do you have a backup of your database?

**Reply by Pouletgrill** [reply]
i don't think i do.. 
I had to setup again the immich admin and all. 
But i do have the Librairy files.

**Reply by bo0tzz** [reply, marked as answer]
Then there is nothing to restore, you'll need to upload the library files again using the CLI.

**Reply by Pouletgrill** [reply]
Alright, how do i  upload the library files again using the CLI ?

**Reply by JW-CH** [reply]
The immich docs provide information about how to use the CLI:
https://immich.app/docs/features/command-line-interface/

**Reply by Pouletgrill** [reply]
ok i will try that and let you know if it went well. 
Thank to you next time i will make sure to backup the database before reinstalling docker if needed.

**Reply by JW-CH** [reply]
@bo0tzz Is it really required to have a backup of the data? Reinstalling docker should not be an issue if his original path is still available (volumes pgdata)

```
volumes:
  pgdata: /old/path/to/pgdata
```

**Reply by Pouletgrill** [reply]
i did not change any path. 
i just reinstalled docker and did a "docker compose up -d" in my immich composer folder.

for the CLI, i try to run this command as mentionned in the documentation, but i get a "Docker invalid reference format." :
`docker run -it -v "$(pwd)":/import:ro -e IMMICH_INSTANCE_URL=http://MY_IP_HERE:2283/api -e IMMICH_API_KEY=API_KEY ghcr.io/immich-app/immich-cli:latest`
keep in mind that i did generate an API Key and put it in the command line, same as the instance_url.

**Reply by Pouletgrill** [reply]
I will post my problem with the CLI as another question. 
https://github.com/immich-app/immich/discussions/8101#discussion-6400196

**Comment by Pouletgrill** [comment]
I manage to made the CLI work with node.js but not with the Docker installation mentionned in the documentation. https://immich.app/docs/features/command-line-interface/#installation-docker
